java-squid/src/main/java/org/sonar/java/resolve/FirstPass.java

```diff
@@ -325,15 +321,19 @@ public void visitEnumConstant(EnumConstantTree tree) {
 
   @Override
   public void visitVariable(VariableTree tree) {
-    declareVariable(computeFlags(tree.modifiers()), tree.simpleName(), (VariableTreeImpl) tree);
+    declareVariable(computeFlags(tree.modifiers(), tree), tree.simpleName(), (VariableTreeImpl) tree);
     super.visitVariable(tree);
   }
 
-  private int computeFlags(ModifiersTree modifiers) {
+  private int computeFlags(ModifiersTree modifiers, Tree tree) {
     int result = 0;
     //JLS7 6.6.1: All members of interfaces are implicitly public.
     if ((env.scope.owner.flags & Flags.INTERFACE) != 0) {
       result = Flags.PUBLIC;
+      if (!tree.is(Tree.Kind.METHOD, Tree.Kind.CONSTRUCTOR)) {
+        // JLS7 9.5: A member type declaration in an interface is implicitly static and public.
```

Comment:
Just to narrow down some tiny details : this is not the only reason as this will also (correctly) handle variable members.